mux.js

```diff
@@ -122,90 +134,112 @@ Multiplexer.prototype._finalize = function(err) {
 // application is allowed to provide optional custom data in the opts for higher-level
 // 'want' selections.
 // The manifest-prop `keys` is required, and must equal an array of strings.
-Multiplexer.prototype.haveFeeds = function (keys, opts) {
+Multiplexer.prototype.offerFeeds = function (keys, opts) {
   var manifest = Object.assign(opts || {}, {
     keys: extractKeys(keys)
   })
   debug(this._id + ' [REPLICATON] sending manifest:', manifest)
-  this._localHave = manifest.keys
+  this._localOffer = this._localOffer.concat(manifest.keys)
```

Comment:
It's *probably* ok, but using `push` throughout to mutate instead of `concat` saves a new list allocation.

Reply:
:+1 

Replacing with 
```js
manifest.keys.forEach(function(key) { this._localOffer.push(key) })
```